lib/mavenlink/request.rb

```diff
@@ -216,7 +216,7 @@ def each_page(batch_size = 200, &block)
       Enumerator.new do |result|
         i = 0
         records_passed = 0
-        request = per_page(batch_size)
+        request = per_page(Mavenlink.specification[collection_name]["per_page"] || batch_size)
```

Comment:
This change will break compatibility with existing usage that passes in a custom `batch_size`. Instead we can maybe default the `batch_size` param to `nil` and then assign it with something like

```
batch_size ||= Mavenlink.specification[collection_name]["per_page"] || DEFAULT_PAGE_LIMIT
```

Reply:
I have updated it as per suggestion.